controllers/twophase/types.go

```diff
@@ -162,15 +162,24 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		chaos.SetNextStart(*nextStart)
 		chaos.SetNextRecover(nextRecover)
 	} else {
+		nextStart, err := utils.NextTime(*chaos.GetScheduler(), status.Experiment.StartTime.Time)
+		if err != nil {
+			r.Log.Error(err, "failed to get next start time")
+			return ctrl.Result{}, err
+		}
 		nextTime := chaos.GetNextStart()
 
-		if !chaos.GetNextRecover().IsZero() && chaos.GetNextRecover().Before(nextTime) {
-			nextTime = chaos.GetNextRecover()
+		if nextStart.Equal(nextTime) {
```

Comment:
Though we don't update the `Scheduler`, we also can't make sure `nextStart` equal to `nextTime'. 

You can check [P144](https://github.com/chaos-mesh/chaos-mesh/blob/master/controllers/twophase/types.go#L144) and [P212](https://github.com/chaos-mesh/chaos-mesh/blob/master/controllers/twophase/types.go#L212). We generate the `nextStart` using `now`, but we set `status.Experiment.StartTime = &metav1.Time{Time: time.Now()}`.  
`now` may not be the same value as `time.Now()`.